### PR TITLE
luhn: update tests and sample implementation

### LINKED ISSUE #245: Changes to Luhn tests
We recently rewrote the test suite for the Luhn exercise. Since this track implements Luhn, please [take a look at the new canonical_data.json file](https://github.com/exercism/x-common/blob/master/exercises/luhn/canonical-data.json) and see if your track should update its implementation or tests.

Details on the test changes and the motivation can be found in [the PR](https://github.com/exercism/x-common/pull/474).

Tracking of these changes is being done at [the tracking issue](https://github.com/exercism/x-common/issues/491).

### PR DESCRIPTION
**Pull request by stkent**
Fixes #245.

Tracking issue: https://github.com/exercism/x-common/issues/491

### CODE REVIEW

**Comment by stkent** [comment]
@jtigger in light of these changes, we should probably re-evaluate our difficulty rating for Luhn.

**Comment by jtigger** [comment]
roger wilco!

**Comment by stkent** [comment]
Updated with the latest and greatest common tests.

In parametrizing tests like this, we lose some ability to describe **why** we are testing the cases listed. Because of that, I'm tempted to remove the parametrization here and instead go back to descriptive test names. (Unless I'm missing a third option which combines the best of both?) Seeking input on this!

**Comment by jtigger** [comment]
Good instincts, @stkent; I agree with 'em... for the reason you state.

In production systems, it seems to serve their role better to have tests be DAMP ([great top-rated answer on StackOverflow](http://stackoverflow.com/questions/6453235/what-does-damp-not-dry-mean-when-talking-about-unit-tests)).

One might say the second you detect a meaningful variance in _why_ a particular pair of input and output are being tested, that's a different test method.

Further, if you're straight-up yearning to say why, well, we're likely quite over that line.

**Comment by stkent** [comment]
Man that answer is fantastic; shared with the team at work today!

I'll refactor these for DAMPness and ping when done.

File: exercises/luhn/src/test/java/LuhnValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+public class LuhnValidatorTest {
+
+    @Test
+    public void testThatSingleDigitStringIsNotValid() {
+        assertFalse(new LuhnValidator().isValid("1"));
+    }
+
+    @Ignore
+    @Test
+    public void testThatTheStringConsistingOfAStringZeroIsInvalid() {
```
Comment (stkent) [inline exercises/luhn/src/test/java/LuhnValidatorTest.java]: Typo: should read "single" zero.

**Comment by stkent** [comment]
Refactored and ready for review!

**Comment by jtigger** [comment]
And I've rescored this exercise, @stkent.

**Pull request merged**